test/extended/util/client.go

```diff
@@ -337,8 +352,155 @@ func (c *CLI) SetupProject() string {
 	return newNamespace
 }
 
+func (c *CLI) setupNamespace() string {
+	requiresTestStart()
+	newNamespace := names.SimpleNameGenerator.GenerateName(fmt.Sprintf("e2e-test-%s-", c.kubeFramework.BaseName))
+	username := fmt.Sprintf("%s-user", newNamespace)
+	serviceAccountName := "default"
+	c.SetNamespace(newNamespace)
+
+	nsObject := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: newNamespace,
+		},
+	}
+	framework.Logf("Creating namespace %q", newNamespace)
+	_, err := c.AdminKubeClient().CoreV1().Namespaces().Create(context.Background(), nsObject, metav1.CreateOptions{})
+	o.Expect(err).NotTo(o.HaveOccurred())
+	c.kubeFramework.AddNamespacesToDelete(nsObject)
+
+	framework.Logf("Waiting for ServiceAccount %q to be provisioned...", serviceAccountName)
+	err = WaitForServiceAccount(c.AdminKubeClient().CoreV1().ServiceAccounts(newNamespace), serviceAccountName)
+	o.Expect(err).NotTo(o.HaveOccurred())
+
+	framework.Logf("Configuring kubeconfig with user %q certificates...", username)
+	c.ChangeUser(username)
```

Comment:
I don't understand how this will work until you modify ChangeUser to tolerate missing the APIs served by the oauth-apiserver and when those APIs are missing use the CSR API to get credentials for the requested user.

Reply:
Found it below